src/extension/src/ProcessHandler.py

```diff
@@ -78,8 +78,8 @@ def start_daemon(self, seq_no, config_settings, ext_env_handler):
         process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         if process.pid is not None:
             self.logger.log("New shell process launched successfully. [Process ID (PID)={0}]".format(str(process.pid)))
-            # Wait for 5 seconds
-            time.sleep(5)
+            # Wait for 2 seconds
+            time.sleep(2)
```

Comment:
Same question from original review -- how was this number arrived at? Why is it not 0, 1, 5, 10? Arbitrary numbers can cause timing issues in production on slower machines than we're used to or due to OS level resource contention.

Reply:
This was based on my local check for now, We had 5 second wait earlier. What I observed with that was, since the core process for assessment generally completes within that time (5 seconds), the handler check on whether the core process is running fails and prints out some of the core logs in the stdout log statement here. The core log file does not have the initial log statements that were recorded within handler logs

Reply:
Please change this to check immediately and in a loop for a fixed large timeout period if process hasn't started yet. The loop can exit when timeout (10 seconds?) expires or the process is created. Intervals between checks should be no larger than the shortest possible legitimate execution time for Core.